packages/nx/src/plugins/js/project-graph/build-dependencies/target-project-locator.ts

```diff
@@ -163,23 +173,27 @@ export class TargetProjectLocator {
 
     try {
       // package.json refers to an external package, we do not match against the version found in there, we instead try and resolve the relevant package how node would
-      const externalPackageJsonPath = findExternalPackageJsonPath(
+      const externalPackageJson = this.findExternalPackageJson(
         packageName,
         fullDirPath
       );
       // The external package.json path might be not be resolvable, e.g. if a reference has been added to a project package.json, but the install command has not been run yet.
-      if (!externalPackageJsonPath) {
+      if (!externalPackageJson) {
+        // Try and fall back to resolving an external node from the graph by name
+        const externalNode = this.npmProjects[`npm:${packageName}`];
+        if (externalNode) {
+          this.npmResolutionCache.set(npmImportForProject, externalNode.name);
```

Comment:
I'm not sure, but I think in the original logic, this also set `undefined` if that's the case? so we don't have to continuously resolve to `undefined`?

Reply:
It set it to null, yeah, will add